unittests/test_parameters.py

```diff
@@ -72,13 +72,26 @@ class MyTest(TwoParams):
 
 def test_filter_params():
     class MyTest(ExtendParams):
-        P1 = parameter(inherit_params=True, filter_params=lambda x: x[2:])
+        # We make the return type of the filtering function a list to ensure
+        # that other iterables different to tuples are also valid.
+        P1 = parameter(inherit_params=True,
+                       filter_params=lambda x: list(x[2:]))
 
     assert MyTest.param_space['P0'] == ('a',)
     assert MyTest.param_space['P1'] == ('d', 'e',)
     assert MyTest.param_space['P2'] == ('f', 'g',)
 
 
+def test_wrong_filter():
+    with pytest.raises(ReframeSyntaxError):
+        class Foo(ExtendParams):
+            P1 = parameter(inherit_params=True, filter_params=lambda x: 1)
+
+    with pytest.raises(ReframeSyntaxError):
+        class Bar(ExtendParams):
+            P1 = parameter(inherit_params=True, filter_params=lambda x, y: [])
```

Comment:
Hmm, I don't think that we document that the returned iterable must not be empty, do we?

Reply:
Oh, the issue here is with the two arguments in the filtering lambda, not the empty iterable on return.

Reply:
I've removed this unit tests for now until the `is_trivially_callable()` becomes available.